src/compiler/tsbuild.ts

```diff
@@ -1182,6 +1183,97 @@ namespace ts {
 
         function buildAllProjects(): ExitStatus {
             if (options.watch) { reportWatchStatus(Diagnostics.Starting_compilation_in_watch_mode); }
+            const originalReadFile = host.readFile;

```

Comment:
how big is the host type? I think it would be less confusing to build a completely new one an temporarily replace the whole thing instead of piecemeal replacing the properties.

Reply:
host has many method + optional ones as well. I think we could do better job (with not having to re-replace them back) once we improve and fix watch to use true form of incremental build and it does its own caching. But for now this is the best thing i think (since watch's incremental stuff uses same host and it wouldnt be correct to use host without correct updates to cache on the changes)